<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>909</width>
    <height>585</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Symulator Przykładowej Maszyny Cyfrowej</string>
  </property>
  <widget class="QWidget" name="mwLayout">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
    <item>
     <widget class="QWidget" name="programWidget" native="true">
      <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0,0">
       <property name="margin">
        <number>0</number>
       </property>
       <item row="4" column="0" colspan="2">
        <layout class="QHBoxLayout" name="programEditLayout">
         <item>
          <widget class="QPushButton" name="programErase">
           <property name="text">
            <string>Wyczyść</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="programSpacer1">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="programDelete">
           <property name="text">
            <string>Usuń</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="programAdd">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Dodaj</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0" colspan="2">
        <widget class="QTableWidget" name="programTable">
         <property name="rowCount">
          <number>0</number>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string extracomment="Dziesiątkowo">Adres</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Zawartość</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Wartość</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <layout class="QHBoxLayout" name="startPCLayout" stretch="2,2,1">
         <item>
          <widget class="QLabel" name="startPCLabel">
           <property name="text">
            <string>Startowa zawartość PC</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="startPCSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLineEdit" name="startPCEdit">
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="runtimeSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget_2">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QTableWidget" name="runtimeTable">
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Adres</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Zawartość</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Interpretacja jako rozkaz</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Interpretacja jako dana</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="logLayout">
          <item>
           <widget class="QLabel" name="logLabel">
            <property name="text">
             <string>Log wykonania</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="logEdit"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="rightLayout" stretch="0,0,0,0,0,0,0,0,0,0,0">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>4</number>
        </property>
        <property name="rightMargin">
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="registersLabel">
          <property name="text">
           <string>Rejestry</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="registersLine">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="registersLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <property name="verticalSpacing">
           <number>6</number>
          </property>
          <item row="5" column="0">
           <widget class="QLabel" name="ORDecLabel">
            <property name="text">
             <string>Dziesiętnie</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLCDNumber" name="ORDecLCD">
            <property name="digitCount">
             <number>16</number>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="ORBinLabel">
            <property name="text">
             <string>Binarnie</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLCDNumber" name="ORBinLCD">
            <property name="numDigits">
             <number>16</number>
            </property>
            <property name="mode">
             <enum>QLCDNumber::Bin</enum>
            </property>
            <property name="segmentStyle">
             <enum>QLCDNumber::Filled</enum>
            </property>
           </widget>
          </item>
          <item row="8" column="0" colspan="2">
           <widget class="Line" name="ACLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="ACDecLabel">
            <property name="text">
             <string>Dziesiętnie</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLCDNumber" name="ACDecLCD">
            <property name="digitCount">
             <number>16</number>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="ACBinLabel">
            <property name="text">
             <string>Binarnie</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QLCDNumber" name="ACBinLCD">
            <property name="numDigits">
             <number>16</number>
            </property>
            <property name="mode">
             <enum>QLCDNumber::Bin</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="Line" name="PCLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="PCDecLabel">
            <property name="text">
             <string>Dziesiętnie</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLCDNumber" name="PCDecLCD">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
              <weight>50</weight>
              <italic>false</italic>
              <bold>false</bold>
             </font>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="lineWidth">
             <number>1</number>
            </property>
            <property name="digitCount">
             <number>16</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="PCLabel">
            <property name="font">
             <font>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Licznik programu</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="ORLabel">
            <property name="text">
             <string>Rejestr operanda</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0" colspan="2">
           <widget class="QLabel" name="ACLabel">
            <property name="text">
             <string>Akumulator</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="rightSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>100</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="controlLabel">
          <property name="text">
           <string>Sterowanie</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="controlLine1">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="verbosityLabel">
          <property name="text">
           <string>Gadatliwość</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="verbositySlider">
          <property name="maximum">
           <number>8</number>
          </property>
          <property name="pageStep">
           <number>2</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="runSppedLabel">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Opóźnienie między instrukcjami</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="runSpeedSlider">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>2000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="controlLayout">
          <item row="0" column="1" colspan="3">
           <widget class="QPushButton" name="controlReset">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="controlSkip">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Następna instrukcja</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QPushButton" name="controlRun">
            <property name="text">
             <string>Wykonuj</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>909</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Plik</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionQuit"/>
   <addaction name="separator"/>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionStart"/>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="exit">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Zakończ</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Zapisz</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Otwórz</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="icon">
    <iconset theme="help">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Pomoc</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>O programie</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Nowy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionStart">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="media-playback-start">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Uruchom symulację</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Zapisz jako</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
